src/2024h2/sandboxed-build-script.md

```diff
@@ -0,0 +1,214 @@
+# Sandboxed build scripts
+
+| Metadata |             |
+| ---------| ----------- |
+| Owner(s) | [weihanglo] |
+| Teams    | [Cargo]     |
+| Status   | WIP         |
+
+[weihanglo]: https://github.com/weihanglo
+[Cargo]: https://www.rust-lang.org/governance/teams/dev-tools#team-cargo
+
+## Summary
+
+Explore different strategies for sandboxing build script executions in Cargo.
+
+## Motivation
+
+Cargo users can opt-in to run [build scripts] in a sandboxed environment,
+limiting their access to OS resources like the file system and network.
+By providing a sandboxed environment for build script executions,
+fewer repetitive code scrutinies are needed.
+The execution of a build script also becomes more deterministic,
+helping the caching story for the ecosystem in the long run.
+
+[build scripts]: https://doc.rust-lang.org/cargo/reference/build-scripts.html
+
+### The status quo
+
+Build scripts in Cargo can do literally anything from network requests to executing arbitrary binaries.
+This isn't deemed a security issue as it is "by design".
+Unfortunately, this "by design" virtue relies on trust among developers within the community.
+When the trust is broken by some incidents,
+even just once,
+the community has no choice but to intensively review build scripts in their dependencies.
+Although there are collaborative code review tools like [cargo-vet] and [cargo-crev] to help build trust,
+that is not realistic and still requires non-negligible efforts on code reviews,
+especially when compared to the speed at which new versions are coming out.
```

Comment:
I think a good way of framing this is that this is like `unsafe`: it will help reviewers direct their efforts to the "here be dragons" parts.

Which means that a good follow up to this would be to allow multiple build scripts in some way so we can better isolate the permissions granted.

Reply:
This is a great idea! Added.